src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/DocumentDbTemplate.java

```diff
@@ -505,15 +505,16 @@ private boolean isCollectionSupportSortByString(@NonNull DocumentCollection coll
     @Override
     public <T> Page<T> findAll(Pageable pageable, Class<T> domainClass, String collectionName) {
         final DocumentQuery query = new DocumentQuery(Criteria.getInstance(CriteriaType.ALL), Sort.unsorted());
-        return paginationQuery(query, pageable, domainClass, collectionName);
+        query.with(pageable);
```

Comment:
can be something like `query = new DocumentQuery(criteria).with(sort).with(pagable)`